courses/machine_learning/deepdive/09_sequence_keras/translate/utils_preproc.py

```diff
@@ -0,0 +1,79 @@
+"""Utilities for preprocessing natural language for machine learning"""
+
+import unicodedata
+import re
+
+import tensorflow as tf
+
+def unicode_to_ascii(s):
+        return ''.join(c for c in unicodedata.normalize('NFD', s)
+            if unicodedata.category(c) != 'Mn')
+
+def preprocess_sentence(w):
+    w = unicode_to_ascii(w.lower().strip())
+
+    # creating a space between a word and the punctuation following it
+    # eg: "he is a boy." => "he is a boy ."
+
+    w = re.sub(r"([?.!,¿])", r" \1 ", w)
+    w = re.sub(r'[" "]+', " ", w)
+
+    # replacing everything with space except (a-z, A-Z, ".", "?", "!", ",")
+    w = re.sub(r"[^a-zA-Z?.!,¿]+", " ", w)
+
+    w = w.rstrip().strip()
+
+    # adding a start and an end token to the sentence
+    # so that the model know when to start and stop predicting.
+    w = '<start> ' + w + ' <end>'
+    return w
+
+def tokenize(lang,lang_tokenizer=None):
```

Comment:
* Add a space after the comma
* "lang" is confusing to me, since it is an abbreviation for language. What about "sentences" instead?
* Why "lang_tokenizer" and not simply "tokenizer"?